### PR TITLE
Update scripts to support automatic updating vesions of spring

### PR DESCRIPTION
**Pull request by Netyyyy**
As title

### CODE REVIEW

File: sdk/spring/scripts/README.md [hunk 1]
```diff
@@ -7,12 +7,9 @@ Scripts in this folder is used to hold the scripts used in Spring Cloud Azure pr
...
-6. Update the comment at the beginning of `\eng\versioning\external_dependencies.txt`: Update the file name of `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`.
-7. Run command `python .\eng\versioning\update_versions.py --ut external_dependency --sr`
-8. Run command `.\eng\versioning\pom_file_version_scanner.ps1 -Debug`. If there is error, fix it.
-9. Update changelog about compatible Spring Boot versions and Spring Cloud versions in `/sdk/spring/CHANGELOG.md`.
-10. Update two spring-boot versions and two spring-cloud versions in`/sdk/spring/ci.yml` for pipeline, then use script `get_spring_boot_managed_external_dependencies.py` generate corresponding file `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`, delete the old two files.
+2. Run command `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py -b SPRING_BOOT_VERSION -c SPRING_CLOUD_VERSION`. Then a file named `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt` will be created in `.\sdk\spring\scripts`.
```
Comment (saragluna) [inline sdk/spring/scripts/README.md:10]: Should we use a placeholder here, like ${SPRING_BOOT_VERSION}

File: sdk/spring/scripts/README.md [hunk 2]
```diff
@@ -7,12 +7,9 @@ Scripts in this folder is used to hold the scripts used in Spring Cloud Azure pr
...
+3. Run command `python .\sdk\spring\scripts\sync_external_dependencies.py -b SPRING_BOOT_VERSION`. Then versions in `\eng\versioning\external_dependencies.txt` will be synchronized with `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`.
```
Comment (saragluna) [inline sdk/spring/scripts/README.md:11]: same here

File: sdk/spring/scripts/README.md [hunk 3]
```diff
@@ -7,12 +7,9 @@ Scripts in this folder is used to hold the scripts used in Spring Cloud Azure pr
...
+4. Update the comment at the beginning of `\eng\versioning\external_dependencies.txt`: Update the file name of `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`.
```
Comment (saragluna) [inline sdk/spring/scripts/README.md:12]: same here

File: sdk/spring/scripts/README.md [hunk 4]
```diff
@@ -7,12 +7,9 @@ Scripts in this folder is used to hold the scripts used in Spring Cloud Azure pr
...
+5. Run command `python .\eng\versioning\update_versions.py --ut external_dependency --sr`
+6. Run command `.\eng\versioning\pom_file_version_scanner.ps1 -Debug`. If there is error, fix it.
+7. Update changelog about compatible Spring Boot versions and Spring Cloud versions in `/sdk/spring/CHANGELOG.md`.
```
Comment (saragluna) [inline sdk/spring/scripts/README.md:15]: we should keep the # 10 here

File: sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py [hunk 5]
```diff
@@ -5,7 +5,8 @@
 #  1. Update `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION` in this script manually.
```
Comment (chenrujun) [inline sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py:5]: This should be updated.

File: sdk/spring/scripts/sync_external_dependencies.py [hunk 6]
```diff
@@ -6,7 +6,8 @@
 #  1. Make sure file(`.\sdk\spring\spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`) exist. If it doesn't exist, please run
 #    `.\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py` to create that file.
 #  2. Update `SPRING_BOOT_VERSION` in this script manually.
```
Comment (chenrujun) [inline sdk/spring/scripts/sync_external_dependencies.py:8]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py [hunk 7]
```diff
@@ -2,10 +2,11 @@
...
 #
 # How to use this script.
-#  1. Update `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION` in this script manually.
+#  1. Get `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION`.
 #     Note that spring-cloud version should compatible with spring-boot version.
 #     Refs: https://spring.io/projects/spring-cloud
-#  2. Run command `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py`.
+#  2. Run command ,sample: `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py -b 2.7.0 -c 2021.0.3`.
+#     Or `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py --spring_boot_dependencies_version 2.7.0 --spring_cloud_dependencies_version 2021.0.3`.
 #  3. Then a file named `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt` will be created.
```
Comment (saragluna) [inline sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py:10]: same here

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]

File: sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py [hunk 8]
```diff
@@ -2,11 +2,12 @@
 # This script is used to get all 3rd party dependencies managed by spring-boot-dependencies and spring-cloud-dependencies.
 #
 # How to use this script.
-#  1. Update `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION` in this script manually.
+#  1. Get `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION`.
 #     Note that spring-cloud version should compatible with spring-boot version.
 #     Refs: https://spring.io/projects/spring-cloud
-#  2. Run command `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py`.
-#  3. Then a file named `spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt` will be created.
+#  2. Run command ,sample: `python .\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py -b 2.7.0 -c 2021.0.3`.
```
Comment (chenrujun) [inline sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py:8]: `Run command ,sample: ` -> `Run command: `

File: sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py [hunk 9]
```diff
@@ -2,11 +2,12 @@
 # This script is used to get all 3rd party dependencies managed by spring-boot-dependencies and spring-cloud-dependencies.
 #
 # How to use this script.
-#  1. Update `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION` in this script manually.
+#  1. Get `SPRING_BOOT_VERSION` and `SPRING_CLOUD_VERSION`.
```
Comment (chenrujun) [inline sdk/spring/scripts/get_spring_boot_managed_external_dependencies.py:5]: Add description about how to get the version, like sdk/spring/scripts/README.md.

File: sdk/spring/scripts/sync_external_dependencies.py [hunk 10]
```diff
@@ -5,8 +5,9 @@
 # How to use this script.
 #  1. Make sure file(`.\sdk\spring\spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`) exist. If it doesn't exist, please run
 #    `.\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py` to create that file.
-#  2. Update `SPRING_BOOT_VERSION` in this script manually.
-#  3. Run command `python .\sdk\spring\scripts\sync_external_dependencies.py`.
+#  2. Get `SPRING_BOOT_VERSION`.
```
Comment (chenrujun) [inline sdk/spring/scripts/sync_external_dependencies.py:8]: 1. Add description about how to get the version. Like sdk/spring/scripts/README.md
2. This should before `Make sure file(`.\sdk\spring\spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`) exist`.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**